### PR TITLE
[OSIDB-2791] [OSIM: Update impact sorting to be logical]

### PR DESCRIPTION
**Pull request by tboyetteredhat**
# [OSIDB-2791] [OSIM: Update impact sorting to be logical]
# [OSIDB-2806] [OSIM: Implement advanced searching on emptiness for CVE ID]

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:
[Update impact sorting to be logical]
[Implement advanced searching on emptiness for CVE ID]
## Changes:
- Update impact sorting to be logical
- Implement advanced searching on emptiness for CVE ID

## Considerations:

Need more support for flaw field emptiness search on advanced search (will cover in different ticket)

### CODE REVIEW

File: src/composables/useFlawAffectsModel.ts [hunk 1]
```diff
@@ -12,12 +12,17 @@ import { useToastStore } from '@/stores/ToastStore';
 import type { ZodFlawType } from '@/types/zodFlaw';
 import type { ZodAffectType, ZodAffectCVSSType } from '@/types/zodAffect';
 import { deepCopyFromRaw } from '@/utils/helpers';
+import { sortWith, ascend, prop } from 'ramda';
+
+const sortAffects = (affects = []) => {
+  return sortWith([ascend(prop('ps_module')), ascend(prop('ps_component'))], affects);
```
Comment (JakubFrejlach) [inline src/composables/useFlawAffectsModel.ts:18]: I think this is not entirely the thing, which is asked for. To sort by product family, you need to use `Affect.ps_product` field which was added to OSIDB API and for example things like rhel-8 and rhel-9 will both fall under Red Hat Enterprise Linux product family. Will send a SFM2 reference in slack.

**Review by JakubFrejlach — CHANGES_REQUESTED** [review]

File: src/composables/useFlawAffectsModel.ts (hunk 1 above)
Comment (tboyetteredhat) [reply src/composables/useFlawAffectsModel.ts:18]: Thanks @JakubFrejlach  I have one question. what happen if ps_product is null, are we showing those affects at the top or bottom in SFM2?

File: src/components/__tests__/FlawForm.spec.ts [hunk 2]
```diff
@@ -614,6 +623,83 @@ describe('FlawForm', () => {
...
+        affectedness: 'AFFECTED',
+        resolution: 'FIX',
+        ps_module: 'shell-4',
+        ps_component: 'openshift',
+        ps_product:'Red Hat Enterprise Linux',
+        impact: 'LOW',
+        // cvss2: '',
+        // cvss2_score: null,
+        // cvss3: '',
+        // cvss3_score: null,
```
Comment (C-Valen) [inline src/components/__tests__/FlawForm.spec.ts:642]: I think if we won't use those we could just omit them. (Same thought for next cases).

File: src/composables/useFlawAffectsModel.ts [hunk 3]
```diff
@@ -12,13 +12,26 @@ import { useToastStore } from '@/stores/ToastStore';
...
 import type { ZodAffectType, ZodAffectCVSSType } from '@/types/zodAffect';
 import { deepCopyFromRaw } from '@/utils/helpers';
-import { equals } from 'ramda';
+import { sortWith, ascend, prop, equals } from 'ramda';
+
+
+const sortAffects = (affects: ZodAffectType[] = []) => {
+  return sortWith(
+    [
+      // get product value for Ramda ascend function
```
Comment (C-Valen) [inline src/composables/useFlawAffectsModel.ts:21]: I'd sightly modify this to also inform about what the function does.
E.g. "get product value using Ramda ascend function as comparator"

File: src/composables/useFlawAffectsModel.ts [hunk 4]
```diff
@@ -12,13 +12,26 @@ import { useToastStore } from '@/stores/ToastStore';
...
+      ascend((affect: ZodAffectType) => affect.ps_product ?? ''),
+      ascend(prop('ps_module')),
+      ascend(prop('ps_module'))
```
Comment (C-Valen) [inline src/composables/useFlawAffectsModel.ts:24]: Is this intentionally repeated?

Comment (tboyetteredhat) [reply src/composables/useFlawAffectsModel.ts]: oh, It should use ps_component.

**Review by C-Valen — APPROVED** [review]
LGTM

File: src/composables/useFlawAffectsModel.ts [hunk 5]
```diff
@@ -12,13 +12,26 @@ import { useToastStore } from '@/stores/ToastStore';
...
+    affects
+  );
+};
 
 export function useFlawAffectsModel(flaw: Ref<ZodFlawType>) {
   const wereAffectsModified = ref(false);
   const modifiedAffectIds = ref<string[]>([]);
   const affectsToDelete = ref<ZodAffectType[]>([]);
-  const theAffects: Ref<ZodAffectType[]> = toRef(flaw.value, 'affects');
+  const theAffects: Ref<ZodAffectType[]> = toRef(sortAffects(flaw.value.affects));
```
Comment (MrMarble) [inline src/composables/useFlawAffectsModel.ts:34]: I think that because of the sort, we should use `ref` instead of `toRef`

**Review by d10n — APPROVED** [review]

**Pull request merged**